excise/tasks.py

```diff
@@ -0,0 +1,63 @@
+import json
+import logging
+
+import opentracing
+import opentracing.tags
+
+from saleor.celeryconf import app
+from saleor.core.taxes import TaxError
+from saleor.order.events import external_notification_event
+from saleor.order.models import Order
+from .utils import AvataxConfiguration, api_post_request, get_metadata_key
+
+logger = logging.getLogger(__name__)
+
+
+@app.task(
+    autoretry_for=(TaxError,), retry_backoff=60, retry_kwargs={"max_retries": 5},
+)
+def api_post_request_task(transaction_url, data, config, order_id):
+    logger.debug("***in the task***")
+    config = AvataxConfiguration(**config)
+    order = Order.objects.filter(id=order_id).first()
+    if not order:
+        logger.error(
+            "Unable to send the order %s to Avatax Excise. Order doesn't exist.",
+            order_id,
+        )
+        return
+    if not data.get("TransactionLines"):
+        msg = "The order doesn't have any line which should be sent to Avatax Excise."
+        external_notification_event(
+            order=order, user=None, message=msg, parameters=None
+        )
+        return
+
+    with opentracing.global_tracer().start_active_span(
+        "avatax_excise.transactions.create"
+    ) as scope:
+        span = scope.span
+        span.set_tag(opentracing.tags.COMPONENT, "tax")
+        span.set_tag("service.name", "avatax_excise")
+
+        response = api_post_request(transaction_url, data, config)
+    msg = f"Order sent to Avatax Excise. Order ID: {order.token}"
+    if not response or "Error" in response.get("Status"):
+        errors = response.get("TransactionErrors", [])
+        avatax_msg = ""
+        for error in errors:
+            avatax_msg += error.get("ErrorMessage", "")
+        msg = f"Unable to send order to Avatax Excise. {avatax_msg}"
+        logger.warning(
+            "Unable to send order %s to Avatax Excise. Response %s",
+            order.token,
+            response,
+        )
+
+    tax_item = {get_metadata_key("itemized_taxes"): json.dumps(response.get("TransactionTaxes"))}
```

Comment:
Why do you store the full API response in metadata? 
Here are more details - https://github.com/saleor/saleor-plugin-avatax-excise/pull/1#discussion_r594238184

Also, one more thing. Our assumption is to keep metadata as a flat object. Storing huge dict object can be problematic, for future processing and  displaying them on the dashboard. 
If you really need to store values received from ATE, you should convert all important fields to flat object and save only them in metadata.

Reply:
```
[
    {
        "TransactionTaxAmounts": [],
        "SequenceId": 1,
        "TransactionLine": 1,
        "InvoiceLine": 1,
        "CountryCode": "USA",
        "Jurisdiction": "TX",
        "LocalJurisdiction": "48",
        "ProductCategory": 0.0,
        "TaxingLevel": "STA",
        "TaxType": "S",
        "RateType": "G",
        "RateSubtype": "NONE",
        "CalculationTypeInd": "P",
        "TaxRate": 0.0625,
        "TaxQuantity": 0.0,
        "TaxAmount": 1.73,
        "TaxExemptionInd": "N",
        "SalesTaxBaseAmount": 27.71,
        "LicenseNumber": "",
        "RateDescription": "TX STATE TAX - TEXAS",
        "Currency": "USD",
        "SubtotalInd": "C",
        "StatusCode": "ACTIVE",
        "QuantityInd": "B"
    },
]
```
It is a response taken from tests.  Do you need all these fields? I mean, to figure out how to properly handle this case, it would be good to check the payload which you want to save.

Reply:
We could filter out just the fields we want for sure. It would still be an array. I can discuss with our client. Given the proximity of our client's launch though I think we should do this after.